pilot/pkg/xds/filters/filters.go

```diff
@@ -306,6 +306,31 @@ var (
 	}
 )
 
+func BuildSidecarOutboundMetadataFilter(stripHeaders bool) *hcm.HttpFilter {
+	return &hcm.HttpFilter{
+		Name: MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
+				map[string]any{
+					"upstream_discovery": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+						map[string]any{
+							"workload_discovery": map[string]any{},
+						},
+					},
+					"upstream_propagation": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+					},
+					"strip_headers": stripHeaders,
```

Comment:
I think this can be done with @howardjohn 's recent pr

Reply:
maybe i am mistaken https://github.com/istio/api/pull/2240/files#diff-5efecd978cf3d28daaa316c3aa309e7987276386df4157cc2b60474e3b7b87ea

Reply:
I don't think @howardjohn's recent change would handle the intelligence of strip x envoy metadata for out of mesh while keeping it for in mesh.

Reply:
added new field to proxyHeaders:
https://github.com/istio/api/pull/2910